### PR TITLE
Add loader for deleting source

### PR DESCRIPTION
**Pull request by rvsia**
**Description**

- Adds loader for deleting a source.
- Also the source is deleted from the application state, so no more refreshing the whole list.

**Before**

deletebefore

**Success**

successdelete

**Error**

error

TODO
- [x] remove double errors
- [x] refactor the list
- [x] Tests

@terezanovotna

### CODE REVIEW

**Comment by terezanovotna** [comment]
I like it!

**Comment by rvsia** [comment]
@Hyperkid123

File: src/api/entities.js [hunk 1]
```diff
@@ -112,14 +112,14 @@ export function doUpdateSource(source, formData) {
             .then((authenticationDataOut) => {
                 return authenticationDataOut;
             }, (_error) => {
-                throw { error: 'Authentication update failure.' };
+                throw { error: { title: 'Authentication update failure.' } };
```
Comment (Hyperkid123) [inline src/api/entities.js]: Its not very good idea to completly discard the error response. If you dont want to show it as notification a least display it in console.

File: src/redux/actions/providers.js [hunk 2]
```diff
@@ -82,17 +82,20 @@ export const updateSource = (source, formData, title, description) => (dispatch)
...
 
-export const removeSource = (sourceId, title) => (dispatch) =>
-    doRemoveSource(sourceId).then(_finished => dispatch({
-        type: ADD_NOTIFICATION,
-        payload: {
-            variant: 'success',
-            title
+export const removeSource = (sourceId, title) => ({
+    type: ACTION_TYPES.REMOVE_SOURCE,
+    payload: () => doRemoveSource(sourceId),
```
Comment (Hyperkid123) [inline src/redux/actions/providers.js]: The promise middleware was deliberately removed in earlier versions of this repo. Are you sure you want it Back?

File: src/api/entities.js (hunk 1 above)
Comment (rvsia) [reply src/api/entities.js]: This was just to preserve compatibility with new notification settings, the whole edit form should be changed at some time in the future :skull_and_crossbones: but I changed it

File: src/redux/actions/providers.js (hunk 2 above)
Comment (rvsia) [reply src/redux/actions/providers.js]: Was there any specific reason? I think it serves its purpose, the code is clean and is officially supported by notifications middleware.

Comment (Hyperkid123) [reply src/redux/actions/providers.js]: It was getting in a way for more complex requests. That is it. You can still use the notification middleware without promise middleware.

**Pull request merged**